### ISSUE TITLE
LNA state set automatically?

### ISSUE THREAD

**Issue by WIHDIA** [softsyst/qirx_issues#81]
Hi Clem,

Most of the time I'm using QIRX and RTL-SDR, but I also like using it with RSP1A. With some strong and weak muxes to listen to I often need to reset the LNA state to avoid overload with RSP1A. To listen to 9B Antenne DE I need to set the LNA state to 4/5 (I also use an external LNA next to the antennas), but switching to local and very strong 12A makes QIRX go berserk unless I set the LNA state to 8 or even 9. 
Would it be possible to automate this like setting the AGC or even choose between automatic or manual setting?

73

Herman
BELGIUM editor [fmlist.org](https://www.fmlist.org)
[My logblog](https://wijnants.info/fmdx/)

**Comment by softsyst** [comment]
Hi Herman,

The problem is well-known. An automatic LNA setting would be most desirable in the RSPs. However, qirx will not try to realize such a thing. 
 
The gain settings in the RSPs are complicated, and setting the LNA state automatically when e.g. an overload happens would be a combined HW-SW procedure causing more than one nightmare, e.g. timing behavior in connection with the scanner.
 
When I started with the V3 of their API, I made some tests to realize it, which failed. 

IMO, this should be integrated in their (sdrplay's) software, just like with the rtl_tcp, to provide the users with a fully automatic AGC, including the LNA state.

73,
Clem
